src/nupic/algorithms/Connections.hpp

```diff
@@ -335,6 +335,13 @@ class Connections : public Serializable
    */
   UInt32 segmentFlatListLength() const;
 
+  /**
+   * Get the vector length needed to use synpase indices.
+   *
+   * @retval A vector length
+   */
+  UInt32 synapseFlatListLength() const;
```

Comment:
the API of Connections is getting bloated, how will this be useful? 
I'd suggest unless there's a good use-case or request for the functionality, make this a private helper method (would be great if more of the methods in Conn can me moved away from public: )

Reply:
synapseFlatListLength() is the size of the vectors needed to store the updates to the synapses.